portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,70 +27,107 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+ 
 /** Servlet that returns a random quote. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
+ 
+    /** Declared as class variable so that initialization happens once in init()
+        rather than everytime in doGET() */
+    List<String> funFacts = new ArrayList<>();
     
-    /** One time initialization may be used for future. */
-    // @Override
-    // public void init() {
-    // }
-
-
-    /** Merges dataFactsAndComments with pastComments to convert as one ArrayList into JSON to put on servlet. */
+    /** One time initialization. */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        List<String> funFacts = new ArrayList<>();
+    public void init() {
         funFacts.add("I enjoy all music genres especially EDM!");
         funFacts.add("I studied abroad in Barcelona :)");
         funFacts.add("My favorite food is rotisserie chicken!");
         funFacts.add("I spend a lot of days on the lake!");
+    }
+ 
+    /** Merges dataFactsAndComments with pastComments to convert as one ArrayList into JSON to put on servlet. */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {        
+        // Add fun facts first to factsAndComments return variable.
+        List<String> factsAndComments = new ArrayList<>();
+        factsAndComments.addAll(funFacts);        
         
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query query = new Query("siteComments").addSort("timeStamp", SortDirection.DESCENDING);
         PreparedQuery results = datastore.prepare(query);
-
+ 
         List<String> pastComments = new ArrayList<>();
+        int validNumRequest = 0;
```

Comment:
It looks like this variable could be scoped to the inside of the for-loop.  It makes the code a bit clearer to read and understand if you instantiate your variables inside the deepest scope possible, that way people reading your code don't have to wonder if this variable is used somewhere after the for loop.

Reply:
Makes sense